### PR TITLE
Record overall Filter latency for all nodes in a scheduling cycle.

### PR DESCRIPTION
**Pull request by liu-cong**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

This is an alternative to #87422 to fix potential p99 predicate evaluation latency overhead caused by extensive metric recording.

Ref #87435

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig scheduling
cc @ahg-g

### CODE REVIEW

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -490,6 +490,15 @@ func (g *genericScheduler) findNodesThatPassFilters(ctx context.Context, state *
 		}
 	}
 
+	beginCheckNode := time.Now()
+	statusCode := framework.Success
+	defer func() {
+		// We record Filter extension point latency here instead of in framework.go because framework.RunFilterPlugins
```
Comment (haosdent) [inline pkg/scheduler/core/generic_scheduler.go]: s/because/because of/

File: pkg/scheduler/framework/v1alpha1/framework.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ const (
 	preFilter                                 = "PreFilter"
 	preFilterExtensionAddPod                  = "PreFilterExtensionAddPod"
 	preFilterExtensionRemovePod               = "PreFilterExtensionRemovePod"
-	filter                                    = "Filter"
+	Filter                                    = "Filter"
```
Comment (haosdent) [inline pkg/scheduler/framework/v1alpha1/framework.go]: Do we have a convention that public variables need to put after private variables?

**Comment by ahg-g** [comment]
Thanks, note that Add/RemovePod PreFilter extensions are also invoked during preemption, but I don't think we should measure them separately, measuring preemption end-to-end (which we already do) just like we are measure Filter end-to-end should be sufficient.

/lgtm
/approve

File: pkg/scheduler/core/generic_scheduler.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/core/generic_scheduler.go]: that is not right.

File: pkg/scheduler/framework/v1alpha1/framework.go (hunk 2 above)
Comment (liu-cong) [reply pkg/scheduler/framework/v1alpha1/framework.go]: not sure. I put it before private ones

**Pull request merged**